### PR TITLE
Add vault lookup, separate cancelation flow

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: src/PoolManager.sol [hunk 1]
```diff
@@ -527,11 +523,11 @@ contract PoolManager is Auth {
     }
 
     function getLiquidityPool(uint64 poolId, bytes16 trancheId, uint128 currencyId) public view returns (address) {
-        return pools[poolId].tranches[trancheId].liquidityPools[currencyIdToAddress[currencyId]];
+        return TrancheTokenLike(pools[poolId].tranches[trancheId].token).vault(currencyIdToAddress[currencyId]);
```
Comment (AStox) [inline src/PoolManager.sol:526]: Bit more convoluted than before, but it seems like a necessary change to conform to ERC7575?

File: src/token/Tranche.sol [hunk 2]
```diff
@@ -32,8 +33,13 @@ contract TrancheToken is ERC20 {
 
     mapping(address => bool) public trustedForwarders;
 
+    /// @dev Look up vault by the asset (part of ERC7575)
+    mapping(address asset => address) public vault;
```
Comment (AStox) [inline src/token/Tranche.sol:37]: I'm not a huge fan of referring to liquidity pools by another name here (although I understand its a requirement for the ERC). Should we consider renaming liquidity pools to vaults?

**Review by AStox — COMMENTED** [review]
I wasnt part of the conversation behind this one, so I'm trying to work backwards to understand this PR. Seems like we're:
1) consolidating the erc7575 interface
2) restructuring how we keep track of liquidity pools in order to conform with erc7575
Is that accurate?

File: src/PoolManager.sol (hunk 1 above)
Comment (hieronx) [reply src/PoolManager.sol:526]: Yes exactly, this is part of ERC7575. I do think this is beneficial for integrations, they can easily find a vault (liquidity pool) for any asset, from the tranche token contract.

**Review by AStox — DISMISSED** [review]

**Review by ilinzweilin — APPROVED** [review]

**Pull request merged**